app/assets/javascripts/discourse/widgets/user-menu.js.es6

```diff
@@ -81,42 +105,39 @@ createWidget("user-menu-links", {
       });
     }
 
+    glyphs.push(this.notificationsGlyph());
     glyphs.push(this.bookmarksGlyph());
 
-    if (siteSettings.enable_personal_messages) {
+    if (this.siteSettings.enable_personal_messages) {
       glyphs.push(this.messagesGlyph());
     }
 
-    if (allowAnon) {
-      if (!isAnon) {
-        glyphs.push({
-          action: "toggleAnonymous",
-          label: "switch_to_anon",
-          className: "enable-anonymous",
-          icon: "user-secret"
-        });
-      } else {
-        glyphs.push({
-          action: "toggleAnonymous",
-          label: "switch_from_anon",
-          className: "disable-anonymous",
-          icon: "ban"
-        });
-      }
-    }
-
-    // preferences always goes last
-    glyphs.push({
-      label: "user.preferences",
-      className: "user-preferences-link",
-      icon: "cog",
-      href: `${path}/preferences`
-    });
-
     return h("ul.menu-links-row", [
       links.map(l => h("li.user", this.linkHtml(l))),
       h("li.glyphs", glyphs.map(l => this.glyphHtml(l)))
     ]);
+  },
+
+  markAsActive(definition) {
+    // Clicking on an active quick access tab icon should redirect the user to
+    // the full page.
+    delete definition.action;
```

Comment:
This is probably not a perf issue here, but in general you don't want to change the shape of the JS objects. https://mathiasbynens.be/notes/shapes-ics

I would recommend setting the values to `null` instead of `delete`.

Reply:
That was a very informative read, TIL! Thanks!